### PR TITLE
ENH add feedstock parsing task to bot

### PR DESCRIPTION
**Pull request by beckermr**
- [x] Pydantic model updated or no update needed

todo

- [x] make extra PRs with random fixes needed
- [x] set defaults for using container to `False`

closes #2261

### CODE REVIEW

File: conda_forge_tick/feedstock_parser.py [hunk 1]
```diff
@@ -452,3 +456,75 @@ def load_feedstock(
...
+            "--cap-drop=all",
+            "--mount",
+            "type=tmpfs,destination=/tmp,tmpfs-mode=1777,tmpfs-size=10000000",  # 10 MB
+            "-m",
+            "2048m",
+            "--cpus",
+            "1",
+            "--rm",
+            "-t",
+            f"{get_default_container_name()}",
```
Comment (jaimergp) [inline conda_forge_tick/feedstock_parser.py:516]:
```suggestion
            get_default_container_name(),
```

File: conda_forge_tick/feedstock_parser.py [hunk 2]
```diff
@@ -452,3 +456,75 @@ def load_feedstock(
...
+            "python",
+            "/opt/autotick-bot/docker/run_bot_task.py",
+            "parse-feedstock",
+            "--existing-feedstock-node-attrs",
+            dumps(sub_graph),
+        ]
+        + args,
+        capture_output=True,
```
Comment (jaimergp) [inline conda_forge_tick/feedstock_parser.py:524]:
```suggestion
        capture_output=True,
        text=True,
```

File: conda_forge_tick/feedstock_parser.py [hunk 3]
```diff
@@ -452,3 +456,75 @@ def load_feedstock(
...
+    )
+    if res.returncode != 0:
+        raise RuntimeError(
+            f"Error running containerized feedstock parsing: {res.stderr}"
+        )
+    return loads(res.stdout.decode("utf-8"))
```
Comment (jaimergp) [inline conda_forge_tick/feedstock_parser.py:530]:
```suggestion
    return loads(res.stdout)
```

File: conda_forge_tick/update_upstream_versions.py [hunk 4]
```diff
@@ -208,6 +214,24 @@ def get_latest_version_containerized(name, attrs, sources):
...
+    sources: Iterable[AbstractSource],
+) -> Dict[str, Union[Literal[False], str]]:
+    """
+    Given a package, return the new version information to be written into the cf-graph.
+
+    :param name: the name of the package.
+    :param attrs: the node attributes of the package
+    :param sources: the version sources to use (sources can be excluded by the package but not added)
+    """
+    if True:
```
Comment (jaimergp) [inline conda_forge_tick/update_upstream_versions.py:229]: This is probably a leftover from debugging or similar?

File: docker/run_bot_task.py [hunk 5]
```diff
@@ -31,7 +32,27 @@
 )
 
 
+@contextmanager
+def _setenv(name, value):
+    """set an environment variable temporarily"""
+    old = os.environ.get(name, None)
```
Comment (jaimergp) [inline docker/run_bot_task.py:38]: `None` is already the default return value, so you don't need it there if you don't want.

File: tests/test_container_tasks.py [hunk 6]
```diff
@@ -47,10 +55,10 @@ def test_container_tasks_version_json():
             "run",
             "--rm",
             "-t",
-            "conda-forge-tick:test",
+            f"{get_default_container_name()}",
```
Comment (jaimergp) [inline tests/test_container_tasks.py:58]:
```suggestion
            get_default_container_name(),
```

File: tests/test_container_tasks.py [hunk 7]
```diff
@@ -73,3 +81,78 @@ def test_get_latest_version_containerized(monkeypatch):
...
+    if "CI" not in os.environ:
+        monkeypatch.setenv("CI", "true", prepend=False)
+
+    res = subprocess.run(
+        [
+            "docker",
+            "run",
+            "--rm",
+            "-t",
+            f"{get_default_container_name()}",
```
Comment (jaimergp) [inline tests/test_container_tasks.py:96]:
```suggestion
            get_default_container_name(),
```

File: tests/test_container_tasks.py [hunk 8]
```diff
@@ -73,3 +81,78 @@ def test_get_latest_version_containerized(monkeypatch):
...
+            attrs = copy.deepcopy(lzj.data)
+            existing_feedstock_node_attrs = dumps(lzj.data)
+
+    res = subprocess.run(
+        [
+            "docker",
+            "run",
+            "--rm",
+            "-t",
+            f"{get_default_container_name()}",
```
Comment (jaimergp) [inline tests/test_container_tasks.py:131]:
```suggestion
            get_default_container_name(),
```

Given how often this type of subprocess is run, I wonder if a helper utility function would help avoid the duplication.

File: conda_forge_tick/update_deps.py [hunk 9]
```diff
@@ -362,7 +362,11 @@ def get_grayskull_comparison(attrs, version_key="version"):
     gs_recipe = make_grayskull_recipe(attrs, version_key=version_key)
 
     # load the feedstock with the grayskull meta_yaml
-    new_attrs = load_feedstock(attrs.get("feedstock_name"), {}, meta_yaml=gs_recipe)
+    new_attrs = load_feedstock(
+        attrs.get("feedstock_name"),
+        {},
+        meta_yaml=gs_recipe,
+    )
```
Comment (beckermr) [inline conda_forge_tick/update_deps.py:369]:
```suggestion
    new_attrs = load_feedstock(attrs.get("feedstock_name"), {}, meta_yaml=gs_recipe)
```

File: conda_forge_tick/feedstock_parser.py [hunk 10]
```diff
@@ -452,3 +461,136 @@ def load_feedstock(
...
+        "-t",
+        get_default_container_name(),
+        "python",
+        "/opt/autotick-bot/docker/run_bot_task.py",
+        "parse-feedstock",
+        "--existing-feedstock-node-attrs",
+        dumps(sub_graph),
+        *args,
+        "--log-level",
+        str(logging.getLevelName(logger.getEffectiveLevel())).lower(),
```
Comment (beckermr) [inline conda_forge_tick/feedstock_parser.py:521]:
```suggestion
        str(logging.getLevelName(logger.getEffectiveLevel())).lower(),
        *args,
```

File: conda_forge_tick/feedstock_parser.py [hunk 11]
```diff
@@ -452,3 +461,137 @@ def load_feedstock(
...
+        *args,
```
Comment (beckermr) [inline conda_forge_tick/feedstock_parser.py:522]:
```suggestion
```

File: conda_forge_tick/feedstock_parser.py [hunk 12]
```diff
@@ -452,3 +456,116 @@ def load_feedstock(
...
+    Returns
+    -------
+    sub_graph : MutableMapping
+        The sub_graph, now updated with the feedstock metadata
+    """
+    in_container = os.environ.get("CF_TICK_IN_CONTAINER", "false") == "true"
+    if use_container is None:
+        use_container = not in_container
+
+    if use_container:
```
Comment (beckermr) [inline conda_forge_tick/feedstock_parser.py:556]:
```suggestion
    if use_container and not in_container:
```

File: conda_forge_tick/feedstock_parser.py [hunk 13]
```diff
@@ -452,3 +456,116 @@ def load_feedstock(
...
+    return data
+
+
+def load_feedstock(
+    name: str,
+    sub_graph: typing.MutableMapping,
+    meta_yaml: Optional[str] = None,
+    conda_forge_yaml: Optional[str] = None,
+    mark_not_archived: bool = False,
+    use_container: bool = None,
```
Comment (beckermr) [inline conda_forge_tick/feedstock_parser.py:524]:
```suggestion
    use_container: bool = False,
```

File: conda_forge_tick/update_upstream_versions.py [hunk 14]
```diff
@@ -215,7 +215,7 @@ def get_latest_version(
     name: str,
     attrs: Mapping[str, Any],
     sources: Iterable[AbstractSource],
-    use_container: bool = False,
+    use_container: bool = None,
```
Comment (beckermr) [inline conda_forge_tick/update_upstream_versions.py:218]:
```suggestion
    use_container: bool = False,
```

**Pull request merged**